tidy3d/components/source.py

```diff
@@ -520,8 +520,8 @@ def _pol_vector(self) -> Tuple[float, float, float]:
         normal_dir[self.injection_axis] = 1.0
         propagation_dir = list(self._dir_vector)
         pol_vector_s = np.cross(normal_dir, propagation_dir)
-        if np.all(pol_vector_s == 0.0):
-            pol_vector_s = np.array((0, 1, 0)) if self.injection_axis == 0 else np.array((1, 0, 0))
+        if self.angle_theta == 0.0 or np.allclose(pol_vector_s, 0.0):
+            pol_vector_s = np.array((1, 0, 0)) if self.injection_axis == 0 else np.array((0, -1, 0))
```

Comment:
This is still not exactly right, I believe `pol_vector_s` should also be rotated by `angle_phi` in the xy plane, i.e. something like:
`pol_vector_s = np.array((np.cos(angle_phi), np.sin(angle_phi), 0)) if self.injection_axis == 0 else np.array((-np.sin(angle_phi), -np.cos(angle_phi), 0))`

Reply:
Actually what I wrote is also wrong, I thought this was in rotated coordinates such that injection axis is z. I think there should be something like `pol_vector_s = self.rotate_points(pol_vector_s, normal_dir, angle=self.angle_phi)`?